discovery_sdk_utils/tests/test_discovery_sdk_utils.py

```diff
@@ -0,0 +1,261 @@
+import pytest
+import sys
+
+from discovery_sdk_utils import find_errors_in_entity_definition
+
+
+class TestEntityDefinitionValidator(object):
+
+    def test_entity_definition_is_dictionary(self):
+        entity_definition = 'this is a string'
+        errors = find_errors_in_entity_definition(entity_definition)
+        assert errors == ['Your ENTITY_DEFINITION must be a dictionary!']
+
+    def test_patterns_property_exists(self):
+        entity_definition = {'missing': 'keys'}
+        errors = find_errors_in_entity_definition(entity_definition)
+        assert errors == ['Your ENTITY_DEFINITION must contain a "patterns" property!']
+
+    def test_patterns_property_has_correct_value(self):
```

Comment:
Since you have semantic function names, it may be better to flesh them out fully rather than have to number them as "2", "3", etc. For example, you could do:

`test_patterns_property_has_correct_value_[type]` where `[type]` is `string`, `list`, `nested_list`, etc.